### PR TITLE
Rise limit for smart update to 2 hours

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| haproxy | passed |
| storage | passed |
| scaling | passed |
| init-deploy | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| scaling-proxysql | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| upgrade | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| security-context | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| scheduled-backup | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scaling | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| upgrade | passed |
| big-data | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| operator-self-healing | passed |
| security-context | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| scheduled-backup | passed |
| demand-backup-encrypted-with-tls | failed |

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-576-3a9ad0c2

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| big-data | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| recreate | passed |
| restore-to-encrypted-cluster | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| demand-backup-encrypted-with-tls | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| security-context | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -208,7 +204,7 @@ func (r *ReconcilePerconaXtraDBCluster) smartUpdate(sfs api.StatefulApp, cr *api
 
 	log.Info(fmt.Sprintf("primary pod is %s", primary))
 
-	waitLimit := 120
+	waitLimit := 2 * 60 * 60 // 2 hours
```
Comment (delgod) [inline pkg/controller/pxc/upgrade.go:207]: I have a concern, If during the update we got `CrashLoopBackOff` we will be inside valid state for 2 hours.

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/pxc/upgrade.go (hunk 1 above)
Comment (blez) [reply pkg/controller/pxc/upgrade.go:207]: Will it help if we add 
```
			if pod.Status.Phase == corev1.PodFailed {
				return errors.Errorf("pod %s is in failed phase", pod.Name)
			}
```
when waiting for pod restart?

Comment (delgod) [reply pkg/controller/pxc/upgrade.go:207]: yes, please

Comment (blez) [reply pkg/controller/pxc/upgrade.go:207]: https://github.com/percona/percona-xtradb-cluster-operator/pull/576/commits/5d105ade00a867151300f2ec3974df96183053a8

**Review by qJkee — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| haproxy | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| upgrade | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

**Review by hors — APPROVED** [review]

**Pull request merged**